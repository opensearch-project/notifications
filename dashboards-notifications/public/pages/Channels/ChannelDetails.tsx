/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { ChannelItemType } from '../../../models/interfaces';
import {
  EuiButton,
  EuiDescriptionList,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHealth,
  EuiSpacer,
  EuiTitle,
} from '@elastic/eui';
import React, { useContext, useEffect, useState } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import {
  ContentPanel,
  ContentPanelActions,
} from '../../components/ContentPanel';
import { CoreServicesContext } from '../../components/coreServices';
import { ModalConsumer } from '../../components/Modal';
import { BREADCRUMBS, ROUTES } from '../../utils/constants';
import { DeleteChannelModal } from './components/modals/DeleteChannelModal';
import { MuteChannelModal } from './components/modals/MuteChannelModal';
import { renderTime } from '../../utils/helpers';

interface ListItemType {
  title: NonNullable<React.ReactNode>;
  description: NonNullable<React.ReactNode>;
}

interface ChannelDetailsProps extends RouteComponentProps<{ id: string }> {}

export function ChannelDetails(props: ChannelDetailsProps) {
  const context = useContext(CoreServicesContext)!;
  const id = props.match.params.id;
  const [channel, setChannel] = useState<ChannelItemType>();

  useEffect(() => {
    context.chrome.setBreadcrumbs([
      BREADCRUMBS.NOTIFICATIONS,
      BREADCRUMBS.CHANNELS,
      {
        text: channel?.name || 'Ops_channel',
        href: `${BREADCRUMBS.CHANNEL_DETAILS.href}/${id}`,
      },
    ]);
    setChannel({
      id,
      name: 'Ops_channel',
      enabled: true,
      type: 'Email',
      allowedFeatures: ['Alerting', 'Reporting'],
      lastUpdatedTime: new Date().getTime(),
      description: 'This group will send to all operational team members.',
      destination: {
        slack: {
          url:
            'https://hooks.slack.com/services/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',
        },
      },
    });
  }, []);

  const channelDescriptionList: Array<ListItemType> = [
    {
      title: 'Channel name',
      description: channel?.name || '-',
    },
    {
      title: 'Description',
      description: channel?.description || '-',
    },
    {
      title: 'Last updated',
      description: renderTime(channel?.lastUpdatedTime || -1),
    },
    {
      title: 'Channel type',
      description: channel?.type || '-',
    },
    {
      title: 'Sender',
      description: 'robot@gmail.com',
    },
    {
      title: 'Default recipients',
      description: 'Team2, cyberadmin@company.com',
    },
    {
      title: 'Email header',
      description: 'Disabled',
    },
    {
      title: 'Email footer',
      description: 'Disabled',
    },
    {
      title: 'Notification sources',
      description: channel?.allowedFeatures.join(', ') || '-',
    },
  ];

  // list is displayed horizontally on the UI, 4 items per row
  // group items into an array of rows
  const channelDescriptionListGroup = channelDescriptionList
    .concat(
      new Array(
        Math.ceil(channelDescriptionList.length / 4) * 4 -
          channelDescriptionList.length
      ).fill(null)
    )
    .reduce(
      (rows: Array<Array<ListItemType>>, item: ListItemType, i: number) => {
        if (i % 4 === 0) rows.push([item]);
        else rows[rows.length - 1].push(item);
        return rows;
      },
      []
    );

  return (
    <>
      <EuiFlexGroup
        alignItems="center"
        gutterSize="m"
        style={{ maxWidth: 1316 }}
      >
        <EuiFlexItem grow={false}>
          <EuiTitle size="l">
            <h1>{channel?.name || '-'}</h1>
          </EuiTitle>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          {channel?.enabled === undefined ? null : channel.enabled ? (
            <EuiHealth color="success">Active</EuiHealth>
          ) : (
            <EuiHealth color="subdued">Muted</EuiHealth>
          )}
        </EuiFlexItem>
        <EuiFlexItem />
        <ModalConsumer>
          {({ onShow }) => (
            <>
              <EuiFlexItem grow={false}>
                <EuiButton
                  size="s"
                  color="danger"
                  onClick={() =>
                    onShow(DeleteChannelModal, {
                      channels: [channel],
                    })
                  }
                >
                  Delete
                </EuiButton>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButton
                  size="s"
                  iconType={channel?.enabled ? 'bellSlash' : 'bell'}
                  onClick={() =>
                    onShow(MuteChannelModal, {
                      channels: [channel],
                      mute: channel?.enabled,
                    })
                  }
                >
                  {channel?.enabled ? 'Mute channel' : 'Unmute channel'}
                </EuiButton>
              </EuiFlexItem>
            </>
          )}
        </ModalConsumer>
      </EuiFlexGroup>

      <EuiSpacer />
      <ContentPanel
        bodyStyles={{ padding: 'initial' }}
        title="Channel configuration"
        titleSize="s"
        panelStyles={{ maxWidth: 1300 }}
        actions={
          <ContentPanelActions
            actions={[
              {
                component: (
                  <EuiButton
                    size="s"
                    href={`#${ROUTES.EDIT_CHANNEL}/${id}?from=details`}
                  >
                    Edit
                  </EuiButton>
                ),
              },
            ]}
          />
        }
      >
        {channelDescriptionListGroup.map((row, rowIndex) => (
          <div key={`channel-description-row-${rowIndex}`}>
            <EuiSpacer />
            <EuiFlexGroup>
              {row.map((item, itemIndex) => (
                <EuiFlexItem key={`channel-description-item-${itemIndex}`}>
                  {item && <EuiDescriptionList listItems={[item]} />}
                </EuiFlexItem>
              ))}
            </EuiFlexGroup>
            <EuiSpacer />
          </div>
        ))}
      </ContentPanel>
    </>
  );
}
