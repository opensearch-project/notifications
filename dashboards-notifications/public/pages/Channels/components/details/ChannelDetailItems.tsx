/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiDescriptionList,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSpacer,
} from '@elastic/eui';
import React from 'react';
import { ListItemType } from '../../types';

interface ChannelDetailItemsProps {
  listItems: Array<ListItemType>;
}

export function ChannelDetailItems(props: ChannelDetailItemsProps) {
  const getPaddedDescriptionList = (list: Array<ListItemType>) => {
    return list
      .concat(
        new Array(Math.ceil(list.length / 3) * 3 - list.length).fill(null)
      )
      .reduce(
        (rows: Array<Array<ListItemType>>, item: ListItemType, i: number) => {
          if (i % 3 === 0) rows.push([item]);
          else rows[rows.length - 1].push(item);
          return rows;
        },
        []
      );
  };

  return (
    <>
      {getPaddedDescriptionList(props.listItems).map((row, rowIndex) => (
        <div key={`channel-description-row-${rowIndex}`}>
          <EuiSpacer size="s" />
          <EuiFlexGroup>
            {row.map((item, itemIndex) => (
              <EuiFlexItem key={`channel-description-item-${itemIndex}`}>
                {item && (
                  <EuiDescriptionList
                    style={{ wordBreak: 'break-word', whiteSpace: 'pre-line' }}
                    listItems={[item]}
                  />
                )}
              </EuiFlexItem>
            ))}
          </EuiFlexGroup>
          <EuiSpacer size="m" />
        </div>
      ))}
    </>
  );
}
