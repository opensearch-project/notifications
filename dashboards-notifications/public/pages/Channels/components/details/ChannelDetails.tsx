/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
  EuiGlobalToastList,
  EuiHealth,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
import _ from 'lodash';
import React, { useContext, useEffect, useState } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { ChannelItemType } from '../../../../../models/interfaces';
import { ContentPanel } from '../../../../components/ContentPanel';
import { CoreServicesContext } from '../../../../components/coreServices';
import { ModalConsumer } from '../../../../components/Modal';
import { ServicesContext } from '../../../../services';
import {
  BREADCRUMBS,
  NOTIFICATION_SOURCE,
  ROUTES,
} from '../../../../utils/constants';
import { renderTime } from '../../../../utils/helpers';
import { ListItemType } from '../../types';
import { MuteChannelModal } from '../modals/MuteChannelModal';
import { ChannelDetailItems } from './ChannelDetailItems';
import { ChannelDetailsActions } from './ChannelDetailsActions';
import { ChannelSettingsDetails } from './ChannelSettingsDetails';

interface ChannelDetailsProps extends RouteComponentProps<{ id: string }> {}

export function ChannelDetails(props: ChannelDetailsProps) {
  const coreContext = useContext(CoreServicesContext)!;
  const servicesContext = useContext(ServicesContext)!;
  const id = props.match.params.id;
  const [channel, setChannel] = useState<ChannelItemType>();
  const [toasts, setToasts] = useState<Toast[]>([]);

  useEffect(() => {
    coreContext.chrome.setBreadcrumbs([
      BREADCRUMBS.NOTIFICATIONS,
      BREADCRUMBS.CHANNELS,
    ]);
    refresh();
  }, []);

  const refresh = async () => {
    servicesContext.notificationService
      .getChannel(id)
      .then(async (response) => {
        if (response.config_type === 'email') {
          const channel = await servicesContext.notificationService.getEmailConfigDetails(
            response
          );
          if (channel.email?.invalid_ids?.length) {
            coreContext.notifications.toasts.addDanger(
              'The sender and/or some recipient groups might have been deleted.'
            );
          }
          return channel;
        }
        return response;
      })
      .then((response) => {
        setChannel(response);
        coreContext.chrome.setBreadcrumbs([
          BREADCRUMBS.NOTIFICATIONS,
          BREADCRUMBS.CHANNELS,
          {
            text: response?.name || '',
            href: `${BREADCRUMBS.CHANNEL_DETAILS.href}/${id}`,
          },
        ]);
      })
      .catch((error) => {
        const newToast: Toast = {
          id: 'channel-not-found-toast',
          title: 'Channel not found',
          color: 'danger',
          iconType: 'alert',
          text: (
            <>
              <EuiText
                size="s"
                style={{
                  fontWeight: 400,
                  color: 'rgb(52, 55, 65)',
                  marginTop: 10,
                  marginBottom: 20,
                }}
              >
                The channel might have been deleted.
              </EuiText>
              <EuiFlexGroup justifyContent="flexEnd">
                <EuiFlexItem grow={false}>
                  <EuiButton size="s" href={`#${ROUTES.NOTIFICATIONS}`}>
                    View Notifications dashboard
                  </EuiButton>
                </EuiFlexItem>
              </EuiFlexGroup>
            </>
          ),
        };
        setToasts([...toasts, newToast]);
      });
  };

  const nameList: Array<ListItemType> = [
    {
      title: 'Channel name',
      description: channel?.name || '-',
    },
    {
      title: 'Description',
      description: channel?.description || '-',
    },
    {
      title: 'Last updated',
      description: renderTime(channel?.last_updated_time_ms || NaN),
    },
  ];

  const sources: Array<ListItemType> = [
    {
      title: 'Notification sources',
      description:
        channel?.feature_list
          .map((source) => _.get(NOTIFICATION_SOURCE, source, '-'))
          .join(', ') || '-',
    },
  ];

  return (
    <>
      <EuiGlobalToastList
        toasts={toasts}
        dismissToast={(removedToast) => {
          setToasts(toasts.filter((toast) => toast.id !== removedToast.id));
        }}
        toastLifeTimeMs={60000}
      />

      <EuiFlexGroup
        alignItems="center"
        gutterSize="m"
        style={{ maxWidth: 1316 }}
      >
        <EuiFlexItem grow={false}>
          <EuiFlexGroup gutterSize="m" alignItems="flexEnd">
            <EuiFlexItem grow={false}>
              <EuiTitle size="l">
                <h1>{channel?.name || '-'}</h1>
              </EuiTitle>
            </EuiFlexItem>
            <EuiFlexItem grow={false} style={{ paddingBottom: 5 }}>
              {channel?.is_enabled === undefined ? null : channel.is_enabled ? (
                <EuiHealth color="success">Active</EuiHealth>
              ) : (
                <EuiHealth color="subdued">Muted</EuiHealth>
              )}
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
        <EuiFlexItem />
        <EuiFlexItem grow={false}>
          {channel && <ChannelDetailsActions channel={channel} />}
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          {channel && (
            <ModalConsumer>
              {({ onShow }) => (
                <EuiButton
                  data-test-subj="channel-details-mute-button"
                  iconType={channel?.is_enabled ? 'bellSlash' : 'bell'}
                  onClick={() => {
                    if (!channel) return;
                    if (channel.is_enabled) {
                      onShow(MuteChannelModal, {
                        selected: [channel],
                        setSelected: (selected: ChannelItemType[]) =>
                          setChannel(selected[0]),
                      });
                    } else {
                      const newChannel = { ...channel, is_enabled: true };
                      servicesContext.notificationService
                        .updateConfig(channel.config_id, newChannel)
                        .then((resp) => {
                          coreContext.notifications.toasts.addSuccess(
                            `Channel ${channel.name} successfully unmuted.`
                          );
                          setChannel(newChannel);
                        });
                    }
                  }}
                >
                  {channel?.is_enabled ? 'Mute channel' : 'Unmute channel'}
                </EuiButton>
              )}
            </ModalConsumer>
          )}
        </EuiFlexItem>
      </EuiFlexGroup>

      <EuiSpacer />
      <ContentPanel
        bodyStyles={{ padding: 'initial' }}
        title="Name and description"
        titleSize="s"
        panelStyles={{ maxWidth: 1300 }}
      >
        <ChannelDetailItems listItems={nameList} />
      </ContentPanel>

      <EuiSpacer />

      <ContentPanel
        bodyStyles={{ padding: 'initial' }}
        title="Configurations"
        titleSize="s"
        panelStyles={{ maxWidth: 1300 }}
      >
        <ChannelSettingsDetails channel={channel} />
      </ContentPanel>

      <EuiSpacer />

      <ContentPanel
        bodyStyles={{ padding: 'initial' }}
        title="Availability"
        titleSize="s"
        panelStyles={{ maxWidth: 1300 }}
      >
        <ChannelDetailItems listItems={sources} />
      </ContentPanel>
    </>
  );
}
