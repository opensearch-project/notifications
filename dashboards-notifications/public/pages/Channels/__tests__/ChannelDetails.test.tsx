/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { fireEvent, render, waitFor } from '@testing-library/react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { RouteComponentProps } from 'react-router-dom';
import { MOCK_DATA } from '../../../../test/mocks/mockData';
import {
  coreServicesMock,
  notificationServiceMock,
} from '../../../../test/mocks/serviceMock';
import { CoreServicesContext } from '../../../components/coreServices';
import { ServicesContext } from '../../../services';
import { ChannelDetails } from '../components/details/ChannelDetails';

describe('<ChannelDetails/> spec', () => {
  configure({ adapter: new Adapter() });

  it('renders the component', () => {
    const props = { match: { params: { id: 'test' } } };
    const utils = render(
      <ServicesContext.Provider value={notificationServiceMock}>
        <CoreServicesContext.Provider value={coreServicesMock}>
          <ChannelDetails {...(props as RouteComponentProps<{ id: string }>)} />
        </CoreServicesContext.Provider>
      </ServicesContext.Provider>
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('renders a specific channel', async () => {
    const props = { match: { params: { id: 'test' } } };
    const notificationServiceMock = jest.fn() as any;
    notificationServiceMock.notificationService = {
      getChannel: async (id: string) => {
        return MOCK_DATA.chime;
      },
    };
    let container = document.createElement('div');

    act(() => {
      ReactDOM.render(
        <ServicesContext.Provider value={notificationServiceMock}>
          <CoreServicesContext.Provider value={coreServicesMock}>
            <ChannelDetails
              {...(props as RouteComponentProps<{ id: string }>)}
            />
          </CoreServicesContext.Provider>
        </ServicesContext.Provider>,
        container
      );
    });
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('handles a non-existing channel', async () => {
    const props = { match: { params: { id: 'test' } } };
    const notificationServiceMock = jest.fn() as any;
    notificationServiceMock.notificationService = {
      getChannel: async (id: string) => {
        throw "non existing channel"
      },
    };
    let container = document.createElement('div');

    act(() => {
      ReactDOM.render(
        <ServicesContext.Provider value={notificationServiceMock}>
          <CoreServicesContext.Provider value={coreServicesMock}>
            <ChannelDetails
              {...(props as RouteComponentProps<{ id: string }>)}
            />
          </CoreServicesContext.Provider>
        </ServicesContext.Provider>,
        container
      );
    });
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('clicks mute button with channel', async () => {
    const props = { match: { params: { id: 'test' } } };
    const notificationServiceMock = jest.fn() as any;
    notificationServiceMock.notificationService = {
      getChannel: async (id: string) => {
        return MOCK_DATA.chime;
      },
      updateConfig: jest.fn(),
    };

    const utils = render(
      <ServicesContext.Provider value={notificationServiceMock}>
        <CoreServicesContext.Provider value={coreServicesMock}>
          <ChannelDetails {...(props as RouteComponentProps<{ id: string }>)} />
        </CoreServicesContext.Provider>
      </ServicesContext.Provider>
    );

    await waitFor(() => {
      utils.getByTestId('channel-details-mute-button').click();
    });
  });

  it('clicks unmute button with channel', async () => {
    const props = { match: { params: { id: 'test' } } };
    const notificationServiceMock = jest.fn() as any;
    const updateConfig = jest.fn(async () => Promise.resolve());
    notificationServiceMock.notificationService = {
      getChannel: async (id: string) => {
        return MOCK_DATA.slack;
      },
      updateConfig,
    };

    const utils = render(
      <ServicesContext.Provider value={notificationServiceMock}>
        <CoreServicesContext.Provider value={coreServicesMock}>
          <ChannelDetails {...(props as RouteComponentProps<{ id: string }>)} />
        </CoreServicesContext.Provider>
      </ServicesContext.Provider>
    );

    await waitFor(() => {
      utils.getByTestId('channel-details-mute-button').click();
    });
  });
});
