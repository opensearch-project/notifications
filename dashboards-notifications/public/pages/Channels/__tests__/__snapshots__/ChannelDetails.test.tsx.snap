// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChannelDetails/> spec handles a non-existing channel 1`] = `
<div>
  <div
    aria-live="polite"
    class="euiGlobalToastList euiGlobalToastList--right"
    role="region"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="max-width: 1316px;"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <h1
            class="euiTitle euiTitle--large"
          >
            -
          </h1>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="padding-bottom: 5px;"
        />
      </div>
    </div>
    <div
      class="euiFlexItem"
    />
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    />
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    />
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="max-width: 1300px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Name and description
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
    />
    <div
      style="padding: initial;"
    >
      <div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Channel name
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Description
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Last updated
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="max-width: 1300px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Configurations
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
    />
    <div
      style="padding: initial;"
    />
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="max-width: 1300px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Availability
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
    />
    <div
      style="padding: initial;"
    >
      <div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Notification sources
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ChannelDetails/> spec renders a specific channel 1`] = `
<div>
  <div
    aria-live="polite"
    class="euiGlobalToastList euiGlobalToastList--right"
    role="region"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="max-width: 1316px;"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <h1
            class="euiTitle euiTitle--large"
          >
            -
          </h1>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="padding-bottom: 5px;"
        />
      </div>
    </div>
    <div
      class="euiFlexItem"
    />
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    />
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    />
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="max-width: 1300px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Name and description
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
    />
    <div
      style="padding: initial;"
    >
      <div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Channel name
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Description
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Last updated
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="max-width: 1300px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Configurations
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
    />
    <div
      style="padding: initial;"
    />
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="max-width: 1300px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Availability
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
    />
    <div
      style="padding: initial;"
    >
      <div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
              style="word-break: break-word; white-space: pre-line;"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Notification sources
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                -
              </dd>
            </dl>
          </div>
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ChannelDetails/> spec renders the component 1`] = `
<div
  aria-live="polite"
  class="euiGlobalToastList euiGlobalToastList--right"
  role="region"
/>
`;
